### PR TITLE
solving add_child error

### PR DESCRIPTION
**Pull request by hazho**
solved issues:
#140
#220

### CODE REVIEW

File: treebeard/mp_tree.py [hunk 1]
```diff
@@ -373,8 +373,11 @@ def process(self):
                       ' increasing the path.max_length property'
                       ' and UPDATE your database'))
         else:
-            # adding the new child as the last one
-            newobj.path = self.node.get_last_child()._inc_path()
+            # adding the new child as the last one after checking if there is a Last Child
+            if self.node.get_last_child():
```
Comment (graingert) [inline treebeard/mp_tree.py:377]: No need to call get_last_child twice

Comment (hazho) [reply treebeard/mp_tree.py:377]: Had been called Only once, is there another call of it that I am not aware of?

Comment (hazho) [reply treebeard/mp_tree.py:377]: my changes can become like:
newobj.path = [self.node.get_last_child()._inc_path() if self.node.get_last_child() else self.node.path + '0001']

Comment (graingert) [reply treebeard/mp_tree.py:377]: That's still two calls...

File: treebeard/mp_tree.py [hunk 2]
```diff
@@ -373,8 +373,11 @@ def process(self):
...
+                newobj.path = self.node.get_last_child()._inc_path()
+            else:
+                newobj.path = self.node.path + '0001'
```
Comment (graingert) [inline treebeard/mp_tree.py:380]:
```suggestion
            last_child = self.node.get_last_child():
            if last_child is not None:
                newobj.path = last_child._inc_path()
            else:
                newobj.path = self.node.path + '0001'
```

Comment (graingert) [reply treebeard/mp_tree.py:380]: but you can actually just use the absence of a child node to determine if a node is a leaf node: https://github.com/django-treebeard/django-treebeard/pull/235/files#diff-35501ef525349cd39e4713d1a9f64a249fa4fbd31d875513e3a15e65988701a2R365

File: treebeard/mp_tree.py (hunk 1 above)
Comment (hazho) [reply treebeard/mp_tree.py:377]: I'm not aware how to make it one, please, feel free to modify

File: treebeard/mp_tree.py [hunk 3]
```diff
@@ -373,8 +373,13 @@ def process(self):
                       ' increasing the path.max_length property'
                       ' and UPDATE your database'))
         else:
-            # adding the new child as the last one
-            newobj.path = self.node.get_last_child()._inc_path()
+            # adding the new child as the last one after checking if there is a Last Child
+            last_child = self.node.get_last_child():
```
Comment (graingert) [inline treebeard/mp_tree.py:377]:
```suggestion
            last_child = self.node.get_last_child()
```

File: treebeard/mp_tree.py [hunk 4]
```diff
@@ -373,8 +373,13 @@ def process(self):
...
+            newobj.path = (
+                last_child._inc_path()
+                if last_child is not None
+                else self.node.path + '0001'
```
Comment (graingert) [inline treebeard/mp_tree.py:381]: I'm not sure where you got this path format from - it should be via `self.node_cls._get_path(self.node.path, newobj.depth, 1)` right?

Comment (graingert) [reply treebeard/mp_tree.py:381]: it looks like what's happening here is the `self.numchild` attribute has been updated but the database doesn't have the child there

**Pull request closed without merging**